handler.py

```diff
@@ -150,17 +82,19 @@ def store_data(event, context):
             ))
         )
     try:
+        key = compose_key(uid=fx_uid, device_id=device_id, service=service)
         req_json = json.loads(event["body"])
-    except ValueError:
+        logger.info("data: {}".format(req_json))
+    except ValueError as ex:
         return dict(
             headers={"Content-Type": "application/json"},
             statusCode=400,
             body=json.dumps(dict(
-                status=200,
-                error="Invalid payload"
+                status=400,
+                error="Invalid payload: {}".format(ex)
             ))
         )
-    s3_data = req_json["data"].encode("utf-8")
+    s3_data = json.dumps(req_json["data"]).encode("utf-8")
```

Comment:
why the json.dumps, isn't this an opaque block of data?

Reply:
S3 requires the data to be character encoded. The demo data I had was JSON based. Going to look at other serialization methods as well.